integrationTests/multiShard/transaction/sendBulkTransactions_test.go

```diff
@@ -0,0 +1,212 @@
+	"encoding/json"
+	"math/big"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/kyber"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/kyber/singlesig"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/stretchr/testify/assert"
+)
+
+var receivedTxsPerShard map[uint32]int
+
+// This will test if txs are sent correctly intr-shard (shard 1 -> shard 1) and cross-shard (shard 1 -> shard 0)
+func TestNode_SendBulkTransactionsAllTransactionsShouldBeSentCorrectly(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	numOfShards := 2
+	nodesPerShard := 3
+	numMetachainNodes := 0
+	firstSkInShard := uint32(0)
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	nodes := integrationTests.CreateNodes(
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	nodes[0] = integrationTests.NewTestProcessorNode(uint32(numOfShards), 0, firstSkInShard, integrationTests.GetConnectableAddress(advertiser))
+	integrationTests.CreateAccountForNodes(nodes)
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Node.Stop()
+		}
+	}()
+
+	numOfCrossShardTxsToSend := 300
+	numOfIntraShardTxsToSend := 400
+	totalNumOfTxs := numOfCrossShardTxsToSend + numOfIntraShardTxsToSend
+	numOfAccounts := 40
+	receivedTxsPerShard = make(map[uint32]int, 0)
+	mutGeneratedTxHashes := sync.Mutex{}
+	//wire a new hook for generated txs on a node in sender shard to populate tx hashes generated
+	for _, n := range nodes {
+		nodeShardId := n.ShardCoordinator.SelfId()
+		if nodeShardId != sharding.MetachainShardId {
+			txPoolRegister(n, &mutGeneratedTxHashes)
+		}
+	}
+
+	// generate a given number of accounts
+	accounts := generateAccounts(numOfAccounts)
+	accountsByShard := make(map[uint32][]keyPair, 0)
+	for _, acc := range accounts {
+		pkBytes, _ := acc.pk.ToByteArray()
+		shardId := nodes[0].ShardCoordinator.ComputeId(state.NewAddress(pkBytes))
+		accountsByShard[shardId] = append(accountsByShard[shardId], acc)
+	}
+
+	var txs []*transaction.Transaction
+
+	// generate cross shard txs from shard 1 to shard 0
+	for len(txs) < numOfCrossShardTxsToSend {
+		// generate a tx from shard 0 to shard 1 and check if signature is correct
+		randomSenderId := getRandomIndex(accountsByShard[1])
+		sender := accountsByShard[1][randomSenderId]
+
+		randomReceiverId := getRandomIndex(accountsByShard[0])
+		receiver := accountsByShard[0][randomReceiverId]
+
+		tx := generateTx(sender.sk, receiver.pk)
+
+		if verifySignature(sender.pk, tx) != nil {
```

Comment:
I think this is redundant. If a tx has an invalid sig it will be discarded by the interceptor.

Reply:
removed